src/plugins/intel_cpu/src/nodes/memory.cpp

```diff
@@ -604,8 +649,85 @@ void MemoryInput::initOptimalPrimitiveDescriptor() {
 
     auto config = selectedPd->getConfig();
     config.outConfs.front().setMemDesc(mem_desc);
-    //bypass any checks, we enforce the child descriptor
+    // bypass any checks, we enforce the child descriptor
     selectedPd->setConfig(config);
+
+    if (haveSubgraph()) {
+        // Adopt parent configuration, avoid to insert reorder before the MemoryInput.
+        std::vector<PortConfig> inConfs;
+        std::vector<Input::InputConfig> graphInputConfig;
+
+        auto mainInputDesc = getParentOutputMemDesc(getParentEdgeAt(0));
+        auto mainInputPrc = mainInputDesc->getPrecision();  // we have to align precision across all the inputs
+
+        inConfs.emplace_back(mainInputDesc);
+        graphInputConfig.emplace_back(node::Input::InputConfig{mainInputDesc, true});
+
+        for (size_t i = 1; i < getParentEdges().size(); i++) {
+            auto desc = getParentOutputMemDesc(getParentEdgeAt(i))->cloneWithNewPrecision(mainInputPrc);
+            inConfs.emplace_back(desc);
+            graphInputConfig.emplace_back(node::Input::InputConfig{desc, true});
+        }
+
+        std::vector<Input::OutputConfig> graphOutputConfig;
+        for (auto&& portConfig : config.outConfs) {
+            auto desc = portConfig.getMemDesc();
+            graphOutputConfig.emplace_back(node::Input::OutputConfig{desc, false});
+        }
+
+        // configure the inner graph to get the information about output memory descriptors
+        subGraph->Init(body, context, graphInputConfig, graphOutputConfig);
+
+        // for the output descriptors, use the configuration of the graph's output nodes
+        auto outputDescriptors = subGraph->getOutputMemoryDescriptors();
+
+        const auto& desc = outputDescriptors.front();
+
+        std::vector<PortConfig> outConfs;
+
+        outConfs.emplace_back(desc, BlockedMemoryDesc::FULL_MASK, 0);  // use the memory from the first input inPlace
+
+        const NodeConfig config(inConfs, outConfs);
+
+        supportedPrimitiveDescriptors.clear();
+        supportedPrimitiveDescriptors.emplace_back(config, impl_desc_type::undef);
+
+        selectPrimitiveDescriptorByIndex(0);
```

Comment:
Above we enforced the output memory descriptor of the subgraph to be the same as in the node, so **we don't need to redefine the node config**. The only thing we really need to do here is inserting a sanity check:
```cpp
    // for the output descriptors, use the configuration of the graph's output nodes
    auto outputDescriptors = m_graph.getOutputMemoryDescriptors();

    const auto& desc = outputDescriptors.front();

    // just a sanity check
    CPU_NODE_ASSERT(desc->isCompatible(*(config.outConfs.front().getMemDesc())), "Unexpected node <-> subgraph output memory descriptor mismatch");
```
Just to make sure that the output memory descriptor has been really enforced.

Reply:
Hi @maxnick , if we enforce memory descriptor here, the subGraph also need to re-init(https://github.com/xipingyan/openvino/blob/7060e852ef2ff532351b3e8421c127515322767f/src/plugins/intel_cpu/src/nodes/memory.cpp#L683).
Because` subGraph->Init` can only call once(second call can trigger crash).
So I removed `MemoryInput::selectOptimalPrimitiveDescriptor()`, and move then to here.

Reply:
We don't need to enforce it exactly here! It's just enforced if you applied the change suggested in https://github.com/openvinotoolkit/openvino/pull/26130#discussion_r1847984999

Reply:
Sorry, updated.